### PR TITLE
(bug) fix file type checking; refactor app frame resize on window resize

### PR DESCRIPTION
**Pull request by tchan247**

### CODE REVIEW

File: client/js/views/folder.react.jsx [hunk 1]
```diff
@@ -93,7 +94,7 @@ var Folder = React.createClass({
     showFiles = showFiles.map(function (file) {
       var fileName = file.filename;
 
-      return <File icon={Tree.getFileType(fileName)} animation={file.style} onClick={function(){this.props.updateCurrentPath(this.props.currentPath === ''? fileName: this.props.currentPath + '/' + fileName)}.bind(context)}>
+      return <File icon={Tree.getFileType(fileName, file.isFolder)} animation={file.style} onClick={function(){this.props.updateCurrentPath(this.props.currentPath === ''? fileName: this.props.currentPath + '/' + fileName)}.bind(context)}>
```
Comment (dxuehu) [inline client/js/views/folder.react.jsx]: can you move these functions out

File: client/js/views/visualize.react.jsx [hunk 2]
```diff
@@ -83,7 +78,7 @@ var Visualize = React.createClass({
   },
 
   getInitialState: function() {
-    return {commits: [], commitIndex: 0, currentPath: '', fileTree: {}, filePaths : {}};
+    return {windowHeight: $(window).height() - 285, commits: [], commitIndex: 0, currentPath: '', fileTree: {}, filePaths : {}};
```
Comment (dxuehu) [inline client/js/views/visualize.react.jsx]: don't need this as a state var

Comment (tchan247) [inline client/js/views/visualize.react.jsx]: window height needs to be state. window.onresize should change the state and re-render the frame size

**Comment by dxuehu** [comment]
works for now, will refactor to pure CSS if we have time (sticky footer)

**Pull request merged**